File: Core/src/main/kotlin/jokatu/game/input/finishstage/EndStageInputDeserialiser.java
package jokatu.game.input.finishstage;

import jokatu.game.input.DeserialisationException;
import jokatu.game.input.SingleKeyInputDeserialiser;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

import java.util.Map;

/**
 * Deserialises {@link EndStageInput}s.  The correct JSON should look like <code>{"end": true}</code>.
 */
@Component
public class EndStageInputDeserialiser extends SingleKeyInputDeserialiser<EndStageInput> {
	@NotNull
	@Override
	public EndStageInput deserialiseSingleValue(@NotNull Map<String, Object> json, @NotNull Object value) throws DeserialisationException {
		return new EndStageInput();
	}

	@NotNull
	@Override
	protected String getKeyName() {
		return "end";
	}
}

File: Core/src/main/kotlin/jokatu/game/input/finishstage/EndStageInputDeserialiser.kt
package jokatu.game.input.finishstage

import jokatu.game.input.DeserialisationException
import jokatu.game.input.SingleKeyInputDeserialiser
import org.springframework.stereotype.Component

/**
 * Deserialises [EndStageInput]s.  The correct JSON should look like `{"end": true}`.
 */
@Component
class EndStageInputDeserialiser : SingleKeyInputDeserialiser<EndStageInput>("end") {
	@Throws(DeserialisationException::class)
	override fun deserialiseSingleValue(json: Map<String, Any>, value: Any): EndStageInput {
		return EndStageInput()
	}
}
